### PR TITLE
Adding missing nuget.config files

### PR DESCRIPTION
**Pull request by AgneLukoseviciute**
Adding nuget.config files for e2e-test-app and integration-test-app. This should hopefully unblock the CI builds.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7544)

### CODE REVIEW

File: .ado/jobs/e2e-test.yml [hunk 1]
```diff
@@ -44,6 +44,24 @@ jobs:
 
       - template: ../templates/prepare-env.yml
 
+      - task: CmdLine@2
```
Comment (dannyvv) [inline .ado/jobs/e2e-test.yml:47]: When this passes and is in to unblock us, I hope we can try to run the build without these changes to see if the nuget.config is enough to get past.

**Review by dannyvv — APPROVED** [review]

File: .ado/jobs/e2e-test.yml (hunk 1 above)
Comment (dannyvv) [reply .ado/jobs/e2e-test.yml:47]: I saw that you tried with only the nuget.config files and ist still failed, another idea might be to place a Nuget.Config file in the root of the repo.

**Comment by AgneLukoseviciute** [comment]
That fixed E2E-, but only partially helped Integration-test. Looks like something's up with dependencies for folly, @dannyvv @asklar any ideas?

**Comment by asklar** [comment]
@AgneLukoseviciute It's unclear why some projects are ok and why some aren't. From what I can see, nuget.org is mentioned in the list of sources used towards the end, so this seems like something is broken with either the nuget task (i.e. it's not using the nuget from VS but instead it's using a custom nuget.exe that is incompatible), or the nuget.exe in VS is broken in recent builds.
Can you try running `nuget sources` before running the build or restores? That seemed to nudge another repo into picking the right nuget.exe. Also might want to look at running [NugetToolInstall](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/nuget?view=azure-devops). If none of this works, please start a thread with VS Discussions or 1ES to figure out why we're still broken despite having a nuget.config.

**Comment by jonthysell** [comment]
Why are we dealing with nuget.exe at all in this yml? We build the app with run-windows, which restores the nuget packages we need.

**Comment by NickGerleman** [comment]
Has anyone tried the workaround mentioned in https://github.com/microsoft/react-native-windows/issues/7436

`Register-PackageSource -Name MyNuGet -Location https://www.nuget.org/api/v2 -ProviderName NuGet -Trusted`

**Comment by AgneLukoseviciute** [comment]
> Has anyone tried the workaround mentioned in #7436
> 
> `Register-PackageSource -Name MyNuGet -Location https://www.nuget.org/api/v2 -ProviderName NuGet -Trusted`

Haven't tried that here yet. I just added a nuget.config in the root of the repo as suggested fix in another thread, if this doesn't work I'll give that a shot.

**Comment by jonthysell** [comment]
I've rekicked this off since GH and ADO were out of sync. Hopefully now we'll get the pass.

**Comment by jonthysell** [comment]
The Folly project is still not seeing boost.

**Comment by dannyvv** [comment]
@jonthysell : The error is in the nuget packages path:
`error: D:\a\1\s\packages\integration-test-app\windows\packages\boost.1.72.0.0\build\boost.targets`
where nuget places the file at: ` Adding package 'boost.1.72.0' to folder 'D:\a\1\s\packages\integration-test-app\packages'`
Due to the new nuget config having:
```
  <config>
    <add key="repositoryPath" value="packages" />
  </config>
```
which should probably be `windows/packages` which makes me confused how in e2e it does resolve to the right path... This nuget config merging is pretty opaque and non-intuitive....

File: packages/e2e-test-app/NuGet.Config [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jonthysell) [inline packages/e2e-test-app/NuGet.Config:1]: I think this file is in the wrong folder. By placing it here, either nuget restore is not seeing this file at all, or nuget restore is placing the nuget packages into the `packages/e2e-test-app/packages` folder when the Folly project is looking at `packages/e2e-test-app/windows/packages`. Move this into the folder alongside the solution file.

File: packages/integration-test-app/NuGet.Config [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jonthysell) [inline packages/integration-test-app/NuGet.Config:1]: Same as above. Move wile into the windows folder alongside the solution file.

**Comment by jonthysell** [comment]
> @jonthysell Jon Thysell (JAUNTY) FTE : The error is in the nuget packages path:
> `error: D:\a\1\s\packages\integration-test-app\windows\packages\boost.1.72.0.0\build\boost.targets`
> where nuget places the file at: ` Adding package 'boost.1.72.0' to folder 'D:\a\1\s\packages\integration-test-app\packages'`
> Due to the new nuget config having:
> 
> ```
>   <config>
>     <add key="repositoryPath" value="packages" />
>   </config>
> ```
> 
> which should probably be `windows/packages` which makes me confused how in e2e it does resolve to the right path... This nuget config merging is pretty opaque and non-intuitive....

Lol, I just saw the same thing. The nuget.config should fix things, it just needs to be put into the windows folder.

File: packages/integration-test-app/NuGet.Config (hunk 3 above)
Comment (AgneLukoseviciute) [reply packages/integration-test-app/NuGet.Config]: done. Thanks!

File: packages/e2e-test-app/windows/NuGet.Config [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <config>
+    <add key="repositoryPath" value="windows/packages" />
```
Comment (jonthysell) [inline packages/e2e-test-app/windows/NuGet.Config:4]: @AgneLukoseviciute  Now that you've moved it you need to change this back to just `packages`.

Comment (jonthysell) [reply packages/e2e-test-app/windows/NuGet.Config]: Nevermind I just went ahead and did it.

**Pull request merged**